src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ConditionalAccess.cs

```diff
@@ -40,84 +38,53 @@ internal BoundExpression RewriteConditionalAccess(BoundConditionalAccess node, b
             var loweredReceiver = this.VisitExpression(node.Receiver);
             var receiverType = loweredReceiver.Type;
 
-            //TODO: if AccessExpression does not contain awaits, the node could be left unlowered (saves a temp),
-            //      but there seem to be no way of knowing that without walking AccessExpression.
-            //      For now we will just check that we are in an async method, but it would be nice
-            //      to have something more precise.
-            var isAsync = _factory.CurrentMethod.IsAsync;
-
             ConditionalAccessLoweringKind loweringKind;
-            // CONSIDER: If we knew that loweredReceiver is not a captured local
-            //       we could pass "false" for localsMayBeAssignedOrCaptured
-            //       otherwise not capturing receiver into a temp
-            //       could introduce additional races into the code if receiver is captured
-            //       into a closure and is modified between null check of the receiver 
-            //       and the actual access.
-            //
-            //       Nullable is special since we are not going to read any part of it twice
-            //       we will read "HasValue" and then, conditionally will read "ValueOrDefault"
-            //       that is no different than just reading both values unconditionally.
-            //       As a result in the case of nullable, not reading captured local through a temp 
-            //       does not introduce any additional races so it is irrelevant whether 
-            //       the local is captured or not.
-            var localsMayBeAssignedOrCaptured = !receiverType.IsNullableType();
-            var needTemp = IntroducingReadCanBeObservable(loweredReceiver, localsMayBeAssignedOrCaptured);
-
-            if (!isAsync && !node.AccessExpression.Type.IsDynamic() && rewrittenWhenNull == null &&
-                (receiverType.IsReferenceType || receiverType.IsTypeParameter() && needTemp))
+            var lowerToTernary = receiverType.IsNullableType() || node.AccessExpression.Type.IsDynamic();
```

Comment:
Does this prevent some optimizations/folding otherwise applicable to conditional access?